install_miniforge.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/sh
+# install_miniforge.sh -- install miniforge on Linux of MacOS
+
+__THIS_PATH="${0}"
+__THIS_FILE="$(basename "${__THIS_PATH}")"
+
+print_usage() {
+    echo "Install Miniforge on Linux or MacOS" 
+    echo "Usage: ${__THIS_FILE} [-h] [Miniforge3.sh arguments]"
+    # echo ""
+    # echo "(*) To install on Windows, uname, curl or wget, and sh must be installed."
+    echo ""
+    echo "If KERNEL_NAME, MARCH, and RELEASE are not specified, on this platform,"
+    echo "the defaults are:"
+    echo ""
+    echo " KERNEL_NAME=$(uname -s)"
+    echo " MARCH=$(uname -m)"
+    echo " RELEASE=latest"
+    echo ""
+    echo "## Examples:"
+    echo "### Download and run Miniforge3.sh interactively:"
+    echo "${__THIS_PATH}"
+    echo ""
+    echo "### Download Linux aarch64 latest and run Miniforge.sh interactively:"
+    echo "KERNEL_NAME=Linux MARCH=aarch64 RELEASE=latest ${__THIS_PATH}"
+    echo ""
+    echo "### Download and run non-interactively (in batch mode):"
+    echo "${__THIS_PATH} -b -p \"${HOME}/miniforge3\""
+    echo ""
+}
+
+main() {
+    for arg in "${@}"; do
+        case "${arg}" in
+            -h|--help)
+                print_usage
+                exit 0
+                ;;
+            test)
+                (set -e -x; run_tests)
+                exit $?
+                ;;
+            --install-mamba)
+                INSTALL_MAMBA=1
+                shift
+                ;;
+        esac
+    done
+    KERNEL_NAME="${KERNEL_NAME:-"$(uname)"}"
+    MARCH="${MARCH:-"$(uname -m)"}"
+    if [ "${MARCH}" = "Darwin" ]; then
+        MARCH="MacOSX"
+    fi
+
+    RELEASE="${RELEASE:-"latest"}"
+
+    MINIFORGE_URL="https://github.com/conda-forge/miniforge/releases/${RELEASE}/download/Miniforge3-${KERNEL_NAME}-${MARCH}.sh"
+    MINIFORGE_SH="Miniforge3.sh"
+    MINIFORGE_SHA256="${MINIFORGE_SH}.sha256"
+    MINIFORGE_SHA256_URL="${MINIFORGE_URL}.sha256"
+    download "${MINIFORGE_URL}" "${MINIFORGE_SH}"
+
+    # TODO FIXME XXX: determine the version number for the .sha256 URL
+    if [ "${RELEASE}" != "latest" ]; then
+        download "${MINIFORGE_SHA256_URL}" "${MINIFORGE_SHA256}"
+        check_sha256 "${MINIFORGE_SHA256}"
+        if [ $? -gt 0 ]; then
+            echo "ERROR: Checksum failed. You might try running this script again."
+            exit 3
+        fi
+    fi
+
+    sh "${MINIFORGE_SH}" "${@}"
+
+    if [ -n "${INSTALL_MAMBA}" ]; then
+        # shellcheck disable=SC1090
+        # shellcheck source=~/.bashrc
+        . "${HOME}/.bashrc"
+        echo PATH="${PATH}"
+        conda install -y mamba -n root -c conda-forge
```

Comment:
```suggestion
        conda install -y mamba -n root
```

Reply:
why would we need to specify conda-forge?

Reply:
If this script is used as a reference, it may be helpful to explicitly specify the channel to install mamba from.
It's probably not (?) necessary to specify the root environment either, but that's how it should be done by users attempting to do the same thing outside of the script. See: #23 "Include mamba in miniforge?"

Reply:
Hi Mark,

I am also working on a 'unified' installer :smile: 

It is also supposed to solve the sizing [issue #23](https://github.com/conda-forge/miniforge/issues/23) but in a more fundamental way ...

Now we have all `Miniconda$$$` artifacts, that could be extended to `Miniconda$$$`, `Nomiconda$$$` and `Maxiconda$$$` artifacts. This way for @isuruf there is no change whatsoever and I presume he will be happy about that, `Nomiconda$$$` can become the 'nominal' (base) version with **full** `conda` (read: also `conda-build`!) AND **full** `mamba` (read: also [boa](https://github.com/TheSnakePit/boa)) and the `Maxiconda$$$` could become the `conda-forge` equivalent of `anaconda` ... (to date, those on `ARM` have no other way to get a full fledged environment then by setting it up by hand ... I know because I am one of them) :smile: 

Also the script plugs the 'install as root' (read : [Installing for multiple users](https://docs.anaconda.com/anaconda/install/multi-user/)) security hole, as it will create a `conda` **user** and **group**, install in there and open up that for the world. (instead of opening the `/home/<installuser>`, or even worse the `/root` to the world)  The install path is then very nicely `/home/conda/miniforge` (for Miniforge$$$), `/home/conda/forge` (for Nomiforge$$$), and `/home/conda/maxiforge` (for Maxiforge$$$) :tada: 

Also, the script is passing through parameters from the 'unified' installer to the underlying installer (don't forget the PyPy guys) :monocle_face: 

The script is not 100% ready yet, need to apply some macOS-fu still, but I would already like to share it with you ... sadly enough Github doesn't allow me to add a `.sh` (or `.zip` for that matter) to this post :cry: The script is not ready enough for a PR, so I did send it to your gmail, hope you don't mind. :wink: 

Let me know what you think about it!